org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/server/actions/JobUtils.java

```diff
@@ -74,7 +80,63 @@ private static IFile getFileFromEditorInput(IEditorInput input) {
     if (input instanceof IFileEditorInput) {
       return ((IFileEditorInput) input).getFile();
     }
-
     return null;
   }
+
+  public static void scheduleAnalysisOfOpenFiles(SonarLintProject project) {
+    Runnable runnable = () -> scheduleAnalysisOfOpenFiles(project.getProject());
+    if (Display.getCurrent() != null) {
+      runnable.run();
+    } else {
+      Display.getDefault().asyncExec(runnable);
+    }
+  }
+
+  public static void scheduleAnalysisOfOpenFiles(List<SonarLintProject> projects) {
+    projects.forEach(JobUtils::scheduleAnalysisOfOpenFiles);
+  }
+
+  public static void scheduleAnalysisOfOpenFilesInBoundProjects(IServer server) {
+    scheduleAnalysisOfOpenFiles(server.getBoundProjects());
+  }
+
+  public static void scheduleAnalysisOfOpenFiles(Job job, List<SonarLintProject> projects) {
+    job.addJobChangeListener(new JobCompletionListener() {
+      @Override
+      public void done(IJobChangeEvent event) {
+        scheduleAnalysisOfOpenFiles(projects);
```

Comment:
You should maybe check result of previous job (especially if it was cancelled). From the JavaDoc: 

> Notification that a job has completed execution, either due to cancelation, successful completion, or failure.  The event status object indicates how the job finished, and the reason for failure, if applicable.